From 5ae1a5cc33d4f5b666f0f8342dbeb322b5f34682 Mon Sep 17 00:00:00 2001
From: Oleksandr Andrushchenko <andr2000@gmail.com>
Date: Sat, 8 Dec 2018 14:22:21 +0200
Subject: [PATCH 2/3] wmt_loader: fix chip ID before initializing the driver

Signed-off-by: Oleksandr Andrushchenko <andr2000@gmail.com>
---
 wmt_loader.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/wmt_loader.c b/wmt_loader.c
index 36ff5a6..b242b41 100644
--- a/wmt_loader.c
+++ b/wmt_loader.c
@@ -68,6 +68,26 @@ int main(int argc, char *argv[])
         return -1;
     }
 
+    iRet = unlink("/dev/wmtWifi");
+    printf("remove wmtWifi returns:%d, errno:%d\n", iRet, errno);
+
+    if((0x0321 == chipId) || (0x0335 == chipId) || (0x0337 == chipId)) {
+        chipId = 0x6735;
+    }
+    if (0x0326 == chipId) {
+        chipId = 0x6755;
+    }
+    if (0x0279 == chipId) {
+        chipId = 0x6797;
+    }
+
+    iRet = ioctl (gLoaderFd, COMBO_IOCTL_MODULE_CLEANUP, chipId);
+    printf("do module cleanup: %d\r\n", chipId);
+    if( iRet < 0 ) {
+        printf("failed to cleanup module \r\n");
+        return -1;
+    }
+
     // do module init 
     iRet = ioctl(gLoaderFd, COMBO_IOCTL_DO_MODULE_INIT, chipId);
     printf("do module init: %d\r\n", chipId);
-- 
2.19.2

