From c2fbafb8140750890c1adcbd62e0d98d76804110 Mon Sep 17 00:00:00 2001
From: Oleksandr Andrushchenko <andr2000@gmail.com>
Date: Sat, 8 Dec 2018 20:50:46 +0200
Subject: [PATCH 3/3] stp: Allow selecting Chip/SOC ID via makefile

Signed-off-by: Oleksandr Andrushchenko <andr2000@gmail.com>
---
 Makefile            | 5 ++++-
 stp_uart_launcher.c | 4 ++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 10a9b80..abafce8 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,7 @@
-CFLAGS += "-D_GNU_SOURCE"
+#CHIPID_CONV ?= 0x7623
+CHIPID_CONV ?= 0x6735
+
+CFLAGS += -D_GNU_SOURCE -DCHIPID_CONV=${CHIPID_CONV}
 
 all:
 	$(CC) $(CFLAGS) stp_uart_launcher.c -lpthread -o stp_uart_launcher
diff --git a/stp_uart_launcher.c b/stp_uart_launcher.c
index 439beaa..5080d25 100644
--- a/stp_uart_launcher.c
+++ b/stp_uart_launcher.c
@@ -1289,7 +1289,7 @@ static int query_chip_id(void)
         // we do not return here, use another way to get chip id
     }
 #else
-    chipId = 0x7623;
+    chipId = CHIPID_CONV;
 #endif
 
 #if 1
@@ -1367,7 +1367,7 @@ static int check_chip_id(void)
 #endif
 
 #else
-    chipId = 0x7623;
+    chipId = CHIPID_CONV;
 #endif
     //open HIF-SDIO
     fdHifsdio = open("/dev/hifsdiod", O_RDWR | O_NOCTTY);
-- 
2.19.2

